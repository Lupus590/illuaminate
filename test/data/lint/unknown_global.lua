-- config: (lint (only var:unknown-global))

-- `a` is local and b is a bound module within this scope
local a
b = 0
print(a, b, c, c_module)

-- getfenv is present in the default (max) global list
print(getfenv)

function a:foo() return self end -- ok
print(self) -- not ok
