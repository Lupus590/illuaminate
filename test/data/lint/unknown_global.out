unknown_global.lua:[6:13-6:13]: Using unknown global "c" [var:unknown-global]
   │
 6 │ print(a, b, c, c_module)
   │             ^
unknown_global.lua:[12:7-12:10]: Using unknown global "self" [var:unknown-global]
    │
 12 │ print(self) -- not ok
    │       ^^^^
No errors and 2 warnings
